### PR TITLE
Add golden end-to-end test generation scenarios

### PR DESCRIPTION
**Pull request by Evangelink**
## Why

The test-generation plugin needs stable end-to-end scenarios that exercise more than isolated test authoring: starting from an untested .NET solution, extending a partial suite without duplication, and safely testing code blocked by ambient time or filesystem dependencies.

## What changed

- Add an internal `scaffold-dotnet-test-project` helper and route zero-to-one .NET test setup through `code-testing-agent`.
- Add a public `testability-obstacle` skill that introduces the smallest behavior-preserving seam and completes the work with deterministic tests.
- Extend the test generator and testability migration agent with explicit scaffolding, gap-discovery, and migrate-through-tests behavior.
- Add self-contained Vally fixtures and powered scenarios for multi-project scaffolding, partial-suite edge discovery, time/filesystem seams, existing seams, static APIs, and off-target routing.

## Design note

`scaffold-dotnet-test-project` is intentionally hidden from direct model invocation. `code-testing-agent` owns user-facing test-generation requests and invokes the helper by name, avoiding a routing collision and keeping the plugin skill menu within its runtime budget. Its behavior is covered through the consumer eval.

## Validation

- `skill-validator check --plugin ./plugins/dotnet-test`
- `python eng/eval-quality/check_eval_quality.py`
- Built all new .NET fixtures and ran the populated xUnit baseline suites

### CODE REVIEW

File: tests/dotnet-test/code-testing-agent/fixtures/csharp-zero-to-one-scaffolding/Commerce.sln [hunk 1]
```diff
@@ -0,0 +1,24 @@
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio Version 17
+VisualStudioVersion = 17.0.31903.59
+MinimumVisualStudioVersion = 10.0.40219.1
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Commerce.Domain", "src\Commerce.Domain\Commerce.Domain.csproj", "{11111111-1111-1111-1111-111111111111}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Commerce.Pricing", "src\Commerce.Pricing\Commerce.Pricing.csproj", "{22222222-2222-2222-2222-222222222222}"
```
[the remark this answers is not in the document]
Comment (Evangelink) [reply tests/dotnet-test/code-testing-agent/fixtures/csharp-zero-to-one-scaffolding/Commerce.sln]: Good catch - fixed in ca8f82d8e. The solution fixture now uses the SDK-style C# project type GUID for both projects.

File: tests/dotnet-test/code-testing-agent/fixtures/csharp-partial-discount-suite/src/DiscountRules.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            return subtotal;
+        }
+
+        return code.ToUpperInvariant() switch
+        {
+            "SAVE10" => decimal.Round(
+                subtotal * 0.9m,
+                2,
+                MidpointRounding.AwayFromZero),
+            "FLAT5" => Math.Max(0, subtotal - 5m),
```
[the remark this answers is not in the document]
Comment (Evangelink) [reply tests/dotnet-test/code-testing-agent/fixtures/csharp-partial-discount-suite/src/DiscountRules.cs]: Fixed in ca8f82d8e. The FLAT5 lower bound is now an explicit decimal literal (`0m`).
